src/jpeg.imageio/jpeginput.cpp

```diff
@@ -453,6 +493,17 @@ JpgInput::read_native_scanline(int subimage, int miplevel, int y, int /*z*/,
         OIIO_DASSERT(m_next_scanline == 0 && current_subimage() == subimage);
     }
 
+    if (m_use_uhdr) {
+        uhdr_raw_image_t* uhdr_raw = uhdr_get_decoded_image(m_uhdr_dec);
+
+        const size_t row_size = uhdr_raw->stride[UHDR_PLANE_PACKED] * 8;
+        unsigned char* top_left = static_cast<unsigned char *>(uhdr_raw->planes[UHDR_PLANE_PACKED]);
+        unsigned char* row_data_start = top_left + row_size * y;
+        memcpy(data, row_data_start, row_size);
+
+        return true;
+    }
+
```

Comment:
This is going to need to be guarded by some `#if` to only compile if the uhdr library is found at build time.